shared/src/platform/context.ts

```diff
@@ -18,6 +18,11 @@ export interface EndpointPair {
     expose: Endpoint
 }
 
+export interface ClosablePair {
+    pair: EndpointPair
+    close: () => void
```

Comment:
I'd make this a Subscription like [@lguychard had suggested](https://sourcegraph.slack.com/archives/CMT39K56Z/p1591355689267900?thread_ts=1591310354.265900&cid=CMT39K56Z) for the [reasons I explained in the same thread](https://sourcegraph.slack.com/archives/CMT39K56Z/p1591702099293700?thread_ts=1591310354.265900&cid=CMT39K56Z)
```suggestion
    /** Closes the endpoint pair. */
    subscription: Unsubscribable
```